specs/Extensions.md

```diff
@@ -0,0 +1,585 @@
+            for (int i = 0; i < extensionsList.Count; ++i)
+            {
+                if (extensionsList[i].Id == entry.Id)
+                {
+                    try
+                    {
+                        await extensionsList[i].RemoveAsync();
+                    }
+                    catch (Exception exception)
+                    {
+                        MessageBox.Show("Failed to remove extension: " + exception);
+                    }
+                    break;
+                }
+            }
+        }
+        await FillViewAsync();
+    }
+
+    private void ToggleExtensionEnabled(object sender, RoutedEventArgs e)
+    {
+        _ = ToggleExtensionEnabledAsync(sender, e);
+    }
+
+    private async System.Threading.Tasks.Task ToggleExtensionEnabledAsync(object sender, RoutedEventArgs e)
+    {
+        ListEntry entry = (ListEntry)ExtensionsList.SelectedItem;
+        List<CoreWebView2BrowserExtension> extensionsList = await m_coreWebView2.Profile.GetBrowserExtensionsAsync();
+        for (int i = 0; i < extensionsList.Count; ++i)
+        {
+            if (extensionsList[i].Id == entry.Id)
+            {
+                try
+                {
+                    await extensionsList[i].SetEnabledAsync(extensionsList[i].IsEnabled ? false : true);
+                }
+                catch (Exception exception)
+                {
+                    MessageBox.Show("Failed to toggle extension enabled: " + exception);
+                }
+                break;
+            }
+        }
+        await FillViewAsync();
+    }
+}
+
+```
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+
+```IDL
+interface ICoreWebView2Profile6;
+interface ICoreWebView2EnvironmentOptions6;
+interface ICoreWebView2BrowserExtensionSetEnabledCompletedHandler;
+interface ICoreWebView2BrowserExtensionRemoveCompletedHandler;
+interface ICoreWebView2BrowserExtension;
+interface ICoreWebView2BrowserExtensionList;
+interface ICoreWebView2ProfileAddBrowserExtensionCompletedHandler;
+interface ICoreWebView2ProfileGetBrowserExtensionsCompletedHandler;
+
+/// Additional options used to create WebView2 Environment.
+[uuid(4B1F63E9-F7A5-4EA5-8D84-2B30F2404E82), object, pointer_default(unique)]
+interface ICoreWebView2EnvironmentOptions6 : ICoreWebView2EnvironmentOptions5 {
+  /// When `AreBrowserExtensionsEnabled` is set to `TRUE`, new extensions can be added to user profile and used.
+  /// `AreBrowserExtensionsEnabled` is default to be `FALSE`, in this case, new extensions can't be installed, and
+  /// already installed extension won't be avaliable to use in user profile.
+  /// See `ICoreWebView2BrowserExtension` for Extensions API details.
+  [propget] HRESULT AreBrowserExtensionsEnabled([out, retval] BOOL* value);
+  /// Sets the `AreBrowserExtensionsEnabled` property.
+  [propput] HRESULT AreBrowserExtensionsEnabled([in] BOOL value);
```

Comment:
Is it okay for multiple WebView2's to share a profile, but have some of the WebViews2's have extensions enabled and some with extensions disabled?